### ISSUE TITLE
test-word reports false positives for names containing the substring 'test'

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1376]
## Problem

`test-word` reports a defect for names that merely *contain* the substring "test" (`latest`, `contest`, `attest`, `protester`), even though the rule is about the *word* "test".

## Root cause

`src/main/resources/org/eolang/lints/tests/test-word.xsl:15`:

```xml
<xsl:for-each select="//o[@name and not(@name='φ') and matches(@name, 'test', 'i') and not(generate-id() = generate-id($top))]">
```

`matches(@name, 'test', 'i')` is a substring search, not a word-boundary match. Verified against real EO: a legitimate test named `+can-use-latest` is flagged with "must not contain the word 'test'".

## Minimal example

```eo
+package x
+architect yegor@256.com
+version 0.0.1
+home https://github.com/objectionary
+spdx SPDX-FileCopyrightText: Copyright (c) 2026 Objectionary.com
+spdx SPDX-License-Identifier: MIT

[] > main
  [] > +can-use-latest
    42 > @
```

`+can-use-latest` contains the substring "test" inside "latest" and is wrongly reported.

## Expected behavior

Match the word "test" at token boundaries, e.g. `matches(@name, '(^|-)test(-|$)', 'i')` (or `\btest\b`), so that `latest`/`contest`/`protester` are allowed.

## Related

- `tests/bad-test-name.xsl` — sibling rule that uses `eo:test-name()` and per-kind prefixes.
- Existing packs under `packs/single/test-word/` use `can-test-add` / `tests-helper` and miss the false-positive case.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+722](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
